flask_iiif/restful.py

```diff
@@ -183,12 +172,18 @@ def get(self, version, uuid, region, size, rotation, quality,
         # Trigger event after proccess the api request
         iiif_after_process_request.send(self, **api_after_request_parameters)
 
-        # Built the after request parameters
-        api_after_request_parameters = dict(
-            mimetype=mimetype,
-            image=to_serve
-        )
+        send_file_kwargs = {'mimetype': mimetype}
+        if 'dl' in request.args:
+            basename = secure_filename(request.args.get('dl', ''))
+            if basename.lower() in {'', '1', 'true'}:
+                basename = uuid
 
-        # Trigger event after proccess the api request
-        iiif_after_process_request.send(self, **api_after_request_parameters)
-        return send_file(to_serve, mimetype=mimetype)
+            attachement_filename = '{0}-{1}-{2}-{3}-{4}.{5}'.format(
```

Comment:
WDYT?

If the `dl` in ['', '1', 'true'] then return our default filename format otherwise let the user choose the filename e.x. ``?dl=foo_1.jpg`` or we can move this to a separated parameter ``custom_filename`` e.x. ``?dl=1&custom_filename=foo_1.jpg`` as @greut suggested

```python
if 'dl' in request.args:
    basename = secure_filename(request.args.get('dl', ''))
    
    if basename.lower() in {'', '1', 'true'}:
        # If it doesn't contain a specific filename just return the default  filename format
        attachment_filename = '{0}-{1}-{2}-{3}-{4}.{5}'.format(
           uuid, region, size, quality, rotation, image_format
         )
     else:
         # return the requested filename
         attachment_filename = basename

    send_file_kwargs.update(
        as_attachment=True,
        attachment_filename=secure_filename(attachment_filename),
    )



```